services/requester/client_handler.go

```diff
@@ -0,0 +1,290 @@
+	logger zerolog.Logger,
+	collector metrics.Collector,
+) (*ClientHandler, error) {
+	remote, err := NewRemote(client, config, signer, logger, blocks, txPool, collector)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ClientHandler{
+		remote:    remote,
+		config:    config,
+		store:     store,
+		blocks:    blocks,
+		receipts:  receipts,
+		logger:    logger,
+		collector: collector,
+	}, nil
+}
+
+func (c *ClientHandler) SendRawTransaction(ctx context.Context, data []byte) (common.Hash, error) {
+	// always use remote client
+	return c.remote.SendRawTransaction(ctx, data)
+}
+
+func (c *ClientHandler) GetBalance(
+	ctx context.Context,
+	address common.Address,
+	height int64,
+) (*big.Int, error) {
+	local, err := c.localClient(height)
+	if err != nil {
+		return nil, err
+	}
+
+	return handleCall(func() (*big.Int, error) {
+		return local.GetBalance(ctx, address, height)
+	}, func() (*big.Int, error) {
+		return c.remote.GetBalance(ctx, address, height)
+	}, c.logger.With().Str("client-call", "get balance").Logger())
+}
+
+func (c *ClientHandler) Call(
+	ctx context.Context,
+	data []byte,
+	from common.Address,
+	height int64,
+) ([]byte, error) {
+	local, err := c.localClient(height)
+	if err != nil {
+		return nil, err
+	}
+
+	return handleCall(func() ([]byte, error) {
+		return local.Call(ctx, data, from, height)
+	}, func() ([]byte, error) {
+		return c.remote.Call(ctx, data, from, height)
+	}, c.logger.With().Str("client-call", "call").Logger())
+}
+
+func (c *ClientHandler) EstimateGas(
+	ctx context.Context,
+	data []byte,
+	from common.Address,
+	height int64,
+) (uint64, error) {
+	local, err := c.localClient(height)
+	if err != nil {
+		return 0, err
+	}
+
+	return handleCall(func() (uint64, error) {
+		return local.EstimateGas(ctx, data, from, height)
+	}, func() (uint64, error) {
+		return c.remote.EstimateGas(ctx, data, from, height)
+	}, c.logger.With().Str("client-call", "estimate gas").Logger())
+}
+
+func (c *ClientHandler) GetNonce(
+	ctx context.Context,
+	address common.Address,
+	height int64,
+) (uint64, error) {
+	local, err := c.localClient(height)
+	if err != nil {
+		return 0, err
+	}
+
+	return handleCall(func() (uint64, error) {
+		return local.GetNonce(ctx, address, height)
+	}, func() (uint64, error) {
+		return c.remote.GetNonce(ctx, address, height)
+	}, c.logger.With().Str("client-call", "get nonce").Logger())
+}
+
+func (c *ClientHandler) GetCode(
+	ctx context.Context,
+	address common.Address,
+	height int64,
+) ([]byte, error) {
+	local, err := c.localClient(height)
+	if err != nil {
+		return nil, err
+	}
+
+	return handleCall(func() ([]byte, error) {
+		return local.GetCode(ctx, address, height)
+	}, func() ([]byte, error) {
+		return c.remote.GetCode(ctx, address, height)
+	}, c.logger.With().Str("client-call", "get code").Logger())
+}
+
+func (c *ClientHandler) GetLatestEVMHeight(ctx context.Context) (uint64, error) {
+	local, err := c.localClient(models.LatestBlockNumber.Int64())
```

Comment:
Just like we do for `SendRawTransaction`, here we also need to use the remote client:
```bash
// always use remote client
```
The local cannot know what is the latest EVM block, only the Cadence script can return that information.